### ISSUE TITLE
Write Documentation for CSR Sparse Matrix Class

### ISSUE THREAD

**Issue by mhig-umich** [bkochuna/ners570f20-Lab06#32]
Provide documentation to user on class attributes and methods.

**Comment by bkochuna** [comment]
The initial issue description is pretty good. However, the metrics by which the the "Definition of Done" may be satisfied are somewhat vague.

The documentation should adhere to some guidelines or standards. Development of documentation standards and guidelines should be a separate issue. The best way to pick a standard is to look at some other C++ project and to mimic their standards and guidelines. I'll create a separate issue for that.

Therefore, I recommend the following changes to the issue description:
> This issue is about creating the documentation for the `SparseMatrix_CSR` class.
>

**Comment by bkochuna** [comment]
I think this should depend on #38

**Comment by littlewatkins** [comment]
Documentation could be done in two ways:
1. Comments in code
2. Markdown file

The comments should be written in a way that allows newcomers to follow along with what is happening throughout the code.

The markdown file will provide more thorough documentation of what the CSR method does and how it does it. It should also contain notes about the code it uses, including classes and methods.

**Comment by knbarton** [comment]
I created a branch containing documentation, CSR_documentation_32. I wasn't sure what all really needs to go in the markdown file, but I made section headers for describing what CSR format is, describing the methods, and including a small example code. I added very short descriptions of the methods, but the rest is empty right now.

**Comment by littlewatkins** [comment]
Oh I'm really sorry @knbarton, I talked to Prof. Kochunas yesterday and he said the documentation just needs to be done in the code itself (ie good comments). I totally didn't think about putting a comment about that into Github. You did a really great job on the code comments so I didn't have anything to add to them. But I wasn't going to close the issue until the methods section was done.

**Comment by knbarton** [comment]
Oh, okay! That's great because I wasn't really looking forward to filling out the documentation. I'll go ahead and delete that branch then.
